File: js/benchmark.js
#!/usr/bin/env node

var fns = require('./functions.js'),
    fs = require('fs'),
    operator = require('./operator.js'),
    path, stats, main, timeStart, result, timeEnd, timeTotal;

var argv = require('optimist')
    .usage('Usage: $0 [--sum] [--iterations <num>] [--fast]')
    .boolean('sum')
    .boolean('fast')
    .default('sum', false)
    .default('iterations', 1)
    .default('fast', false)
    .argv;

var problemList;
if (argv.fast) {
    problemList = [1, 2, 3, 5, 6, 8, 13, 15, 16, 18, 19, 20, 24, 28, 40, 45];
} else {
    problemList = operator.range(1, 50 + 1);
}

var j;
for (var i = 0, problemNumber; problemNumber = problemList[i]; i++) {
    path = './complete/no' + fns.zeroPad(problemNumber, 3) + '.js';
    try {
        stats = fs.statSync(path);
    } catch (e) {
        path = './too_slow/no' + fns.zeroPad(problemNumber, 3) + '.js';
    }    

    main = require(path);

    timeStart = new Date();
    for (j = 0; j < argv.iterations; j++) {
        result = main.main();
    }
    timeEnd = new Date();
    // Time in milliseconds
    timeTotal = timeEnd - timeStart;

    if (!argv.sum) {
        timeTotal = Math.floor(timeTotal / argv.iterations);
    }

    console.log(problemNumber + ': ' + result + ', ' + timeTotal + 'ms');
}

File: python/benchmark.py
#!/usr/bin/env python

import argparse
import os
import time

parser = argparse.ArgumentParser()
parser.add_argument('--iterations', metavar='N', type=int,
                    help='number of times to run the benchmark')
parser.add_argument('--sum', action='store_true',
                    help='sum the times (default: find the average)')
parser.add_argument('--fast', action='store_true',
                    help='just execute the short running problems')
args = parser.parse_args()
iterations = 1 if args.iterations is None else args.iterations

if args.fast:
    problem_list = [1, 2, 3, 5, 6, 8, 13, 15, 16, 18, 19, 20, 24, 28, 40, 45]
else:
    problem_list = range(1, 50 + 1)

for problem_number in problem_list:
    path = 'complete/no%03d.py' % problem_number
    if os.path.exists(path):
        module = 'complete.no%03d' % problem_number
    else:
        module = 'too_slow.no%03d' % problem_number

    main = __import__(module, fromlist=['main'])
    begin = time.time()
    for i in range(iterations):
        result = main.main()
    end = time.time()

    total = end - begin
    if not args.sum:
        total = total * 1.0 / iterations

    print '%s: %s, %sms' % (problem_number, result, int(1000 * total))
